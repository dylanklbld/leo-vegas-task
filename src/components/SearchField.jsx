import "../styles/searchfield.css"

import React, { useState } from 'react'

import useDebounce from '../hooks/useDebounce'
import useEffectExceptMount from '../hooks/useEffectExceptMount'

const emptyString = ''

export const SimpleSearchField = ({ onSearchValueChanged }) => {
    const [searchValue, setSearchValue] = useState(emptyString)
    const debouncedSearchValue = useDebounce(searchValue, 500)

    useEffectExceptMount(() => {
        debouncedSearchValue &&
            onSearchValueChanged(debouncedSearchValue)
    }, [debouncedSearchValue])

    return <React.Fragment>
        <div className="leo-task-search-field">
            <span>search icon</span>
            <input type="text" className="leo-task-search-input-field" placeholder="Search for a movie..." value={searchValue} onChange={(e) => {
                setSearchValue(e.target.value)
            }} />
            <div></div>
        </div>
    </React.Fragment>
}
