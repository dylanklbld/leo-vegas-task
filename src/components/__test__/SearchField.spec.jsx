import "@testing-library/jest-dom"

import { fireEvent, render, wait, waitForDomChange } from '@testing-library/react'

import React from 'react'
import { SimpleSearchField } from '../SearchField'

it('should render', () => {
    const { container } = render(<SimpleSearchField onSearchRequestDone={() => { }} />)

    expect(container).toMatchSnapshot()
})


it('should call onSearchValueChanged', async () => {
    const onSearchValueChanged = jest.fn().mockName('onSearchValueChanged')
    const { container } = render(<SimpleSearchField onSearchValueChanged={onSearchValueChanged} />)

    expect(container).toMatchSnapshot()
    await wait()
    const inputField = container.querySelector('.leo-task-search-input-field')
    fireEvent.change(inputField, { target: { value: 'Color' } })

    await new Promise(resolve => setTimeout(resolve, 600))

    expect(onSearchValueChanged).toHaveBeenCalledWith("Color")
})