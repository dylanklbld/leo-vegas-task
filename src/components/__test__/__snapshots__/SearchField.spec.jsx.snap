// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onSearchValueChanged 1`] = `
<div>
  <div
    class="leo-task-search-field"
  >
    <span>
      search icon
    </span>
    <input
      class="leo-task-search-input-field"
      placeholder="Search for a movie..."
      type="text"
      value=""
    />
    <div />
  </div>
</div>
`;

exports[`should render 1`] = `
<div>
  <div
    class="leo-task-search-field"
  >
    <span>
      search icon
    </span>
    <input
      class="leo-task-search-input-field"
      placeholder="Search for a movie..."
      type="text"
      value=""
    />
    <div />
  </div>
</div>
`;
